src/main/java/org/ethereum/beacon/discovery/util/Functions.java

```diff
@@ -15,153 +12,154 @@
-    Sign.SignatureData signatureData =
-        Sign.signMessage(x.toArray(), ECKeyPair.create(key.toArray()), false);
-    Bytes r = Bytes.wrap(signatureData.getR());
-    Bytes s = Bytes.wrap(signatureData.getS());
-    return Bytes.concatenate(r, s);
-  }
-
-  public static void setSkipSignatureVerify(boolean skipSignatureVerify) {
-    SKIP_SIGNATURE_VERIFY = skipSignatureVerify;
+  public static Bytes sign(final SecretKey secretKey, final Bytes32 messageHash) {
+    final KeyPair keyPair = KeyPair.fromSecretKey(secretKey);
+    final Signature signature = SECP256K1.signHashed(messageHash, keyPair);
+    // cutting v
+    return signature.bytes().slice(0, 64);
   }
 
   /**
    * Verifies that signature is made by signer
    *
-   * @param signature Signature, ECDSA
-   * @param x message, hashed
+   * @param signature Signature, ECDSA (r || s parts only, without v, 64 bytes)
+   * @param hashedMessage message, hashed
    * @param pubKey Public key of supposed signer, compressed, 33 bytes
    * @return whether `signature` reflects message `x` signed with `pubkey`
    */
-  public static boolean verifyECDSASignature(Bytes signature, Bytes x, Bytes pubKey) {
-    Preconditions.checkArgument(pubKey.size() == 33, "Invalid public key size");
-
-    if (SKIP_SIGNATURE_VERIFY) {
-      return true;
-    }
-    ECPoint ecPoint = Functions.publicKeyToPoint(pubKey);
-    Bytes pubKeyUncompressed = Bytes.wrap(ecPoint.getEncoded(false)).slice(1);
-    ECDSASignature ecdsaSignature =
-        new ECDSASignature(
-            new BigInteger(1, signature.slice(0, 32).toArray()),
-            new BigInteger(1, signature.slice(32).toArray()));
+  public static boolean verifyECDSASignature(
+      final Bytes signature, final Bytes32 hashedMessage, final Bytes pubKey) {
+    final PublicKey publicKey = derivePublicKeyFromCompressed(pubKey);
     try {
-      for (int recId = 0; recId < 4; ++recId) {
-        BigInteger calculatedPubKey = Sign.recoverFromSignature(recId, ecdsaSignature, x.toArray());
-        if (calculatedPubKey == null) {
-          continue;
-        }
-        if (Arrays.areEqual(
-            pubKeyUncompressed.toArray(),
-            extractBytesFromUnsignedBigInt(calculatedPubKey, PUBKEY_SIZE))) {
-          return true;
-        }
+      final boolean verifyV1 =
+          SECP256K1.verifyHashed(
+              hashedMessage,
+              Signature.create(
+                  (byte) 1,
+                  signature.slice(0, 32).toUnsignedBigInteger(),
+                  signature.slice(32).toUnsignedBigInteger()),
+              publicKey);
+      if (verifyV1) {
+        return true;
       }
-    } catch (final IllegalArgumentException e) {
+      return SECP256K1.verifyHashed(
+          hashedMessage,
+          Signature.create(
+              (byte) 0,
+              signature.slice(0, 32).toUnsignedBigInteger(),
+              signature.slice(32).toUnsignedBigInteger()),
+          publicKey);
```

Comment:
It would be good to deduplicate these two checks. The only difference should be the `v` value so could either extract a method that takes pubKey, signature, message and v value or just use a `for (byte v = 0; v <= 1; v++)` loop.